pyrsp/rsp.py

```diff
@@ -45,15 +45,10 @@ def __init__(self, port):
         self.__dict__['port'] = serial.Serial(port, 115200, timeout=1)
 
     def setup(self, rsp):
-        rsp.send(b'qRcmd,737764705f7363616e')
-        pkt=rsp.readpkt()
-        while pkt!=b'OK':
-            if pkt[:1]!=b'O':
-                raise ValueError('not O: %s' % pkt)
-            if rsp.verbose:
-                print(unhex(pkt[1:-1]))
-            pkt=rsp.readpkt()
+        rsp.command(b'swdp_scan')
         rsp.fetchOK(b'vAttach;1',b'T05')
+        if isinstance(rsp, CortexM3):
+            rsp.command(b'vector_catch enable hard int bus stat chk nocp mm reset')
```

Comment:
is there any reason this needs to be here, and not in the connect() of cortexM3 class?

Reply:
ahh, is this because it is blackmagic specific and not openocd? hmmmm, but then this is a bug for openocd, since it will still trigger if it is a cortexm3

Reply:
Yes, you could say `if isinstance(self.port, BlackMagic): self.command(b'vector_catch enable hard int bus stat chk nocp mm reset')` in CortexM3.connect(), but since this command is BlackMagic-specific (i.e. you find it in the BlackMagic documentation, not in the CortexM3 documentation), it felt more natural to keep it in the BlackMagic class.

It won't trigger for OpenOCD, since OpenOCD implements RSP over TCP, so it uses the (currently perhaps not perfectly named) STlink2 class as its probe.

I don't think these commands (that effectively map to what `gdb monitor` does) are really standardized. Instead, every probe implements their own command set, and we can't force OpenOCD to implement the exact same commands as BlackMagic.